workflows/differentialabundance.nf

```diff
@@ -174,6 +221,15 @@ workflow DIFFERENTIALABUNDANCE {
         ch_features = Channel.of([ exp_meta, file(params.features, checkIfExists: true)])
     } else if (params.study_type == 'affy_array'){
         ch_features = ch_affy_platform_features
+    } else if (params.study_type == 'maxquant'){
+
+        // For maxquant, we will use the processed matrices from PROTEUS
+        ch_features = ch_in_norm
```

Comment:
This is very similar to what happens in the final part of this conditional, so let's combine the two rather than duplicating what happens there. Right now the last bit is

```
        // Otherwise we can just use the matrix input
        matrix_as_anno_filename = "matrix_as_anno.${matrix_file.getExtension()}"
        matrix_file.copyTo(matrix_as_anno_filename)
        ch_features = Channel.of([ exp_meta, file(matrix_as_anno_filename)])
```

But you could do something like:

```
        // Otherwise we can just use the matrix input

        matrix_as_anno_filename = "matrix_as_anno.${matrix_file.getExtension()}"
        if (params.study_type == 'maxquant'){
            ch_features_matrix = ch_in_norm
        } else{
            ch_features_matrix = ch_in_raw
        }
        ch_features = ch_features_matrix
            .map{ exp_meta, matrix_file -> 
                matrix_file.copyTo(matrix_as_anno_filename)
                return [exp_meta, file(matrix_as_anno_filename)]
            }
```

Reply:
TIL that the inside of the map is not separate from the rest of the code. I had to rename `exp_meta, matrix_file` and  `return [exp_meta` to not receive this error:
`- cause: The current scope already contains a variable of the name exp_meta`

Reply:
yep, my bad